### PR TITLE
Test setup

### PR DESCRIPTION
**Pull request by brkidder**
@d-rowe @cadecameron sorry this is pretty late

Major changes
1. Added in the tests to check for simple rendering
2. Refactored to use import throughout module
3. Refactored to use Deal component

Minor changes
1. Dependencies
2. Script
3. Gitignore
You know, the fun stuff.

Thank you for your time.

### CODE REVIEW

File: client/src/components/deal.jsx [hunk 1]
```diff
@@ -1,32 +1,29 @@
...
+      <span>
+        {' $' + deal.salePrice}
+      </span>
+    </p>
+    <p>
+      <span>{deal.bought + '+ bought '}</span>
+      <span>{deal.salePercent + '% OFF'}</span>
+    </p>
+  </label>
+);
```
Comment (d-rowe) [inline client/src/components/deal.jsx]: Nice work refactoring this

File: spec/components/deal.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    extraPrice: '576.10',
+    bought: 140,
+    allowed: 340,
+    soldOut: true,
+    salePercent: 6,
+    extraPercent: 6,
+    packageId: 4,
+    createdAt: '2020-01-28T22:22:24.000Z',
+    updatedAt: '2020-01-28T22:22:24.000Z',
+  };
```
Comment (d-rowe) [inline spec/components/deal.test.js]: Not sure this data is needed for this specific test. I believe that shallow rendering without any deal prop would result the same.

**Comment by d-rowe** [comment]
Good work. I like the refactors you made.

File: spec/components/deal.test.js (hunk 2 above)
Comment (brkidder) [reply spec/components/deal.test.js]: Thank you for taking a look at this. I put in the data in order to write better test in the future. You are right for this specific test I didn't have to put this in

File: client/src/components/deal.jsx (hunk 1 above)
Comment (brkidder) [reply client/src/components/deal.jsx]: thank you

File: client/src/components/deals.jsx [hunk 3]
```diff
@@ -1,38 +1,19 @@
...
-  // `;
-  props.deals.map((deal) => (
-    <div key={deal.id}>
-      <label>
-        <input
-          id={deal.id}
-          type="radio"
-          name="packageDeals"
-          value={deal.title}
+  props.deals.map((deal, i) => {
```
Comment (cadecameron) [inline client/src/components/deals.jsx]: Yeah, great work on pulling the map function back into the `deals.jsx` file @brkidder !

File: package.json [hunk 4]
```diff
@@ -4,11 +4,18 @@
...
+    "test": "jest",
+    "test:watch": "jest --watch --coverage --colors",
+    "test:coverage": "jest --coverage --colors",
     "start": "nodemon server",
     "build": "webpack -d -w",
     "seed": "cd db/ && npx sequelize db:drop && npx sequelize db:create && npx sequelize db:migrate && npx sequelize db:seed:all || npm run fallback",
-    "fallback": "cd db/ && npx sequelize db:create && npx sequelize db:migrate && npx sequelize db:seed:all"
+    "fallback": "cd db/ && npx sequelize db:create && npx sequelize db:migrate && npx sequelize db:seed:all",
+    "jest": {
+      "transform": {
```
Comment (cadecameron) [inline package.json]: consider moving this into the `jest.config.json` file (though it can be in either).

**Pull request merged**